src/lib/validations/sql.ts

```diff
@@ -335,7 +341,7 @@ const checks: Checks = {
 		recommendation: '',
 	},
 	siteHomeUrlLando: {
-		matcher: "'(siteurl|home)',\\s?'([^']+)'",
+		matcher: `['"](siteurl|home)['"],\\s?['"]([^'"]+)['"]`,
```

Comment:
```suggestion
		matcher: `['"](siteurl|home)['"],\\s?(?:['"])([^'"]+)\1`,
```
I wonder if we could make use of back-references here to make the regex more reliable. 

(I haven't tested this regex. )

Reply:
It didn't work - the following test value is supposed to return `true`.

```js
const testValue = 'INSERT INTO `wp_options` VALUES(1,"siteurl","https://6462.abdullaha8ctest.blog","yes")'

// PR's regex
const regexO2 = /['"](siteurl|home)['"],\s?['"]([^'"]+)['"]/

// Suggested regex
const regexN2 = /['"](siteurl|home)['"],\s?(?:['"])([^'"]+)\1/

regexO2.test(testValue) // true

regexN2.test(testValue) // false
```

Reply:
```
const regexN2 = /(['"])(siteurl|home)\1,\s?\1([^'"]+)\1/

regexN2.test(testValue)
```
Although I agree it introduces another capture group.

Reply:
Ah okay, I get it. Took a good look at what it does - it utilizes the type of quote that first matches.

Though thinking hard about it - I'll need to wait until we see what the edge case looks like.

I'm thinking of a scenario where there'll be both an escaped single and double quote - which even then, both regexes will fail - so might be worth deferring until we see it hitting an edge case.